templates/api-server.json

```diff
@@ -0,0 +1,57 @@
+{
+  "name": "api-server",  
+  "description": "A LoopBack API server with local User auth",
+  "package": {
+    "version": "1.0.0",
+    "main": "server/server.js",
+    "scripts": {
+      "lint": "eslint .",
+      "start": "node .",
```

Comment:
@superkhau @deepakrkris could you please summarise the outcome of the discussion on Slack, so that people reading this patch can access it too, even if they don't have access to our Slack?

Also I see that this pull request was already merged. If we are going to continue this discussion, then please open a new GH issue where we can leave our comments.

> do you like the idea of having the template as a json file. the original workspace had it as a folder with .js files

This is not only about JSON vs JS extension. The more important question is whether we want the templates to contain content as it will be scaffolded (i.e. `server/middleware.json` in the current template), or whether we want templates to contain values for Domain Entities provided by workspace?

The former has the benefit that templates are easy to maintain and it's immediately clear what will be written to the filesystem.

The latter has the benefit that a single template can support different configuration formats. For example, we have changed the way how remote methods are defined in LoopBack 3.0. At the same time, we improved loopback-workspace to support both flavours in a single version. That way a single `ModelMethod` template can be used to scaffold the relevant entry in `common/models/my-model.json` regardless of what LoopBack version the app is using. If we were using the first approach, then we would have to maintain multiple versions of the json template (LB 1-2.x, LB 3.x+).